meal-planner/src/hooks/useRecipe.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react'
+import mealPlan from '../data/MealPlan'
+import loadLocally from '../lib/loadLocally'
+import saveLocally from '../lib/saveLocally'
+
+export default function useRecipe() {
+    const [weekdays, setWeekdays] = useState([])
+    const [selectedWeekday, setSelectedWeekday] = useState('')
+    const [showRecipe, setShowRecipe] = useState(false)
+
+    const [selectedMeal, setSelectedMeal] = useState({
+        title: '',
+        instructions: '',
+        ingredients: '',
+        image: '',
+    })
+
+    useEffect(() => {
+        setWeekdays(loadLocally('weekdays') ?? mealPlan)
+    }, [])
```

Comment:
If you want to load weekdays on start, you can remove this `useEffect` and update your `useState`. This will avoid that `weedays` is always an empty array on first render.

```suggestion
    const [weekdays, setWeekdays] = useState(() => loadLocally('weekdays') ?? mealPlan)
    const [selectedWeekday, setSelectedWeekday] = useState('')
    const [showRecipe, setShowRecipe] = useState(false)

    const [selectedMeal, setSelectedMeal] = useState({
        title: '',
        instructions: '',
        ingredients: '',
        image: '',
    })
```

Reply:
sounds reasonable.